Kitodo-API/src/main/java/org/kitodo/serviceloader/KitodoServiceLoader.java

```diff
@@ -52,20 +74,242 @@ public KitodoServiceLoader(Class clazz, String modulePath) {
+     * frontend files
+     */
+    private void loadBeans() {
+        try {
+
+            Path moduleFolder = FileSystems.getDefault().getPath(modulePath);
+            DirectoryStream<Path> stream = Files.newDirectoryStream(moduleFolder, "*.jar");
+
+            for (Path f : stream) {
+                JarFile jarFile = new JarFile(f.toString());
+
+                if (hasFrontendFiles(jarFile)) {
+
+                    Enumeration<JarEntry> e = jarFile.entries();
+
+                    URL[] urls = {new URL("jar:file:" + f.toString() + "!/")};
+                    URLClassLoader cl = URLClassLoader.newInstance(urls);
+
+                    while (e.hasMoreElements()) {
+                        JarEntry je = e.nextElement();
+
+                        /*
+                         * IMPORTANT: Naming convention: the name of the java class has to be in upper
+                         * camel case or "pascal case" and must be equal to the file name of the
+                         * corresponding facelet file concatenated with the word "Form".
+                         *
+                         * Example: template filename "sample.xhtml" => "SampleForm.java"
+                         *
+                         * That is the reason for the following check (e.g. whether the JarEntry name
+                         * ends with "Form.class")
+                         */
+                        if (je.isDirectory() || !je.getName().endsWith("Form.class")) {
+                            continue;
+                        }
+
+                        String className = je.getName().substring(0, je.getName().length() - 6);
+                        className = className.replace('/', '.');
+                        Class c = cl.loadClass(className);
+
+                        String beanName = className.substring(className.lastIndexOf(".") + 1).trim();
+
+                        FacesContext facesContext = FacesContext.getCurrentInstance();
+                        HttpSession session = (HttpSession) facesContext.getExternalContext().getSession(false);
+
+                        session.getServletContext().setAttribute(beanName, c.newInstance());
+                    }
+                }
+            }
+        }
+        catch (Exception e) {
+            logger.error("Classpath could not be accessed", e.getMessage());
+        }
+    }
+
+    /**
+     * If the found jar files have frontend files, they will be extracted and copied into the frontend folder
+     * of the core module. Before copying, existing frontend files of the same module will be deleted from the
+     * core module. Afterwards the created temporary folder will be deleted as well.
+     */
+    private void loadFrontendFilesIntoCore() {
+        Path moduleFolder = FileSystems.getDefault().getPath(modulePath);
+
+        try {
+            DirectoryStream<Path> stream = Files.newDirectoryStream(moduleFolder, "*.jar");
```

Comment:
Class `DirectoryStream` implements the `Closeable` / `Autocloseable` interface which means that we can use the try-with-resources-statement here. This stream object is not closed in current implementation which is an other good reason to change it.